### ISSUE TITLE
2.0.0-beta.1 requires a Github login but works without it in 0.16.54

### ISSUE THREAD

**Issue by Ogglas** [jspm/jspm-cli#2429]
Running the new _jspm 2.0.0-beta.1_ requires a login to Github like the screenshot below when running the `jspm install` command:

![image](https://user-images.githubusercontent.com/4015237/56796974-f5dbc980-6813-11e9-9f61-e6b15f642850.png)

This is not required when running _0.16.54_ . Why is that? This is causing trouble on our Jenkins build server.

**Comment by guybedford** [comment]
This should not be the case. And we shouldn’t with unnecessarily either.
Have you installed any GitHub packages? Or is there a dependency of a
dependency on github? Perhaps share your jspm.json file here if you can.

**Comment by Ogglas** [comment]
Happens using Node.js latest LTS version on Windows 10 and Server 2016. There are Github packages but _jspm install_ does not require login using _0.16.54_.

https://nodejs.org/en/download/

```
PS C:\Users\oscar\Documents\Test> npm -v
6.4.1
PS C:\Users\oscar\Documents\Test> node -v
v10.15.3
```

[package.zip](https://github.com/jspm/jspm-cli/files/3126359/package.zip) contains `config.js` and `package.json` samples.

**Comment by guybedford** [comment]
Thanks, yes I've seen this too. It was quite a while ago I wrote the authentication logic, but I believe it might be trying to login by default? Although I seem to recall that it stores your credentials once you enter them in so that you only ever need to do this once (eg I never see this prompt anymore unless I use jspm on a new machine).

Are you clicking cancel?

As a result is it showing this every time?

Definitely, let's improve this UX.

**Comment by Ogglas** [comment]
Why is a login needed in the new version and not in the old version? It will cause trouble when setting up new build nodes for build servers like TeamCity and Jenkins. For AppVeyor I think you will need to set the login via a script. They don't allow direct access to the build nodes as far as I know.

**Comment by javadbat** [comment]
yeah we have the same problem with docker

**Comment by guybedford** [comment]
I think it is worthwhile to have the prompt for the credentials by default when accessing GitHub as it will avoid ratelimiting.

You can disable this by setting the `GIT_TERMINAL_PROMPT=0` environment variable on CI machines.

Marking as a documentation issue to track a page for listing CI configuration information.

**Comment by guybedford** [comment]
(and let me know if you disagree too!)

**Comment by javadbat** [comment]
is there any command to set this credential before we execute "jspm i"?

**Comment by guybedford** [comment]
`GIT_TERMINAL_PROMPT=0 jspm i` should work? Is that not enough?

**Comment by guybedford** [comment]
Is there a common way to determine if an application is running in a CI environment? Are there standard environment variable checks that can work between them?

If so, we could possibly add support for such a thing here too.

**Comment by guybedford** [comment]
Ok I'm also struggling with this workflow :) Marking as a bug again and will see what progress I can make soon.

**Comment by javadbat** [comment]
have something like this will do the job:
jspm github-credential --user javadbat  --token edjfhkfhkwfdskjfsdkfsk343242897492

**Comment by guybedford** [comment]
Ok so the issue here is that jspm 2.0 is checking whether a GitHub repo is a redirect through the GitHub API, while jspm previously did this more sneakily by just checking `https:` redirects on the repo.

The API check is more correct as that will work with private repos etc, but what I think we can do here instead is to handle GitHub auth failure with a fallback that can attempt the https route, and only provide the auth failure at that point.

Will keep this open to track that work further.

In the mean time, the recommended workflow would be to set the auth token through `JSPM_GITHUB_AUTH_TOKEN`.

**Comment by guybedford** [comment]
This will now work for CI, but should still give the login prompt for normal installs. The benefit of the login is that you get dependency preloading so it makes installs faster.

**Comment by guybedford** [comment]
This was released in 2.0.0-beta.4.
